### PR TITLE
Add utility functions for the Optional type.

### PR DESCRIPTION
**Pull request by hakujin**
Adds two of `Data.Maybe`'s utility functions to `Optional`.  I'm happy to add equivalents for `catMaybe` and `mapMaybe` if you feel they're in scope for this library. Apologies about the whitespace.

### CODE REVIEW

File: src/Data/Optional.hs [hunk 1]
```diff
@@ -147,3 +149,18 @@ instance Fractional a => Fractional (Optional a) where
...
+-- the default value. Otherwise, it applies the function to the value inside the
+-- 'Optional' and returns the result.
+optional :: b -> (a -> b) -> Optional a -> b
+optional n _ Default  = n
+optional _ f (Specific x) = f x
+
+-- | The 'fromOptional' function takes a default value and an 'Optional'
+-- value.  If the 'Optional' is 'Default', it returns the default value;
+-- otherwise, it returns the value contained in the 'Optional'.
+fromOptional :: a -> Optional a -> a
```
Comment (Gabriella439) [inline src/Data/Optional.hs]: How about naming this `defaultTo` instead?

I would like to reserve `fromOptional` for this function:

``` haskell
fromOptional :: Alternative f => Optional a -> f a
fromOptional  Default     = empty
fromOptional (Specific x) = pure x
```

**Comment by hakujin** [comment]
Done. 

As a lone data point, `defaultTo` isn't an immediately obvious description of that function to me. Since we're straying from the `Data.Maybe` conventions, perhaps consider Scala's `getOrElse`? `specificOrElse`?

Will you elaborate on your thought process behind that name?

**Comment by Gabriella439** [comment]
The purpose of the function is to default an `Optional` to some value.  So if you write `defaultTo 1` it reads literally as "default to 1"

**Comment by hakujin** [comment]
Makes sense in that context. Thanks.

**Pull request merged**